### DISCUSSION TITLE
[Docs] 📕 进阶教程 - 利用 Cloudflare Zero Trust Tunnel 实现 CodeSandbox 自定义域名

### DISCUSSION THREAD

**Discussion by Harry-zklcdc** [Harry-zklcdc/go-proxy-bingai#338] [Show and tell]
> [!NOTE]
>
> 在正式按照教程进行操作之前, 请先确保域名绑定至 Cloudflare

## 1. 注册 Cloudflare Zero Trust

具体注册方法请出门右拐隔壁百度 or Google

> 绑定银行卡时, 可以使用 PayPal 来跳过绑定银行卡的步骤

## 2. 创建 Tunnel

进入 Zero Trust 后台

<img width="1415" alt="image-20240205195226139" src="https://github.com/Harry-zklcdc/go-proxy-bingai/assets/21104213/563d13fe-d516-411f-bb76-9bcf29a2503a">

选择「Network」-> 「Tunnel」-> 「Create a Tunnel」

<img width="1415" alt="image-20240205195548836" src="https://github.com/Harry-zklcdc/go-proxy-bingai/assets/21104213/8bc9a75a-590f-49ee-8c7e-34d925c38d23">

选择「Cloudflared」

<img width="1415" alt="image-20240205195724206" src="https://github.com/Harry-zklcdc/go-proxy-bingai/assets/21104213/b1a382cc-63d6-4332-bef1-679642b55824">

填入该隧道的注释名称

<img width="1415" alt="image-20240205195803078" src="https://github.com/Harry-zklcdc/go-proxy-bingai/assets/21104213/2f72de16-fcaf-470f-aa4d-8592e006f707">

点击右边方框内的内容, 复制安装命令

<img width="1415" alt="image-20240205195858664" src="https://github.com/Harry-zklcdc/go-proxy-bingai/assets/21104213/39490105-0f4e-41c4-9920-eca05390164d">
类似

```bash
sudo cloudflared service install eyJhIjoiM2ZjNjQ4xxxxxxxxxxxxxxx
```

这时候, 将后面的 `eyJhIjoiM2ZjNjQ4xxxxxxxxxxxxxxx` 复制出来保存

### ~~HuggingFace~~

> [!WARNING]
>
> HuggingFace 已经封杀本项目, 本部分无需再看

<details><summary>点击查看内容</summary>

> [!NOTE]
>
> 需要重新进行 [HuggingFace 部署](https://github.com/Harry-zklcdc/go-proxy-bingai/wiki/HuggingFace-%E9%83%A8%E7%BD%B2), 以更新代码支持

将 `eyJhIjoiM2ZjNjQ4xxxxxxxxxxxxxxx` 填入 HuggingFace Space 的机密中

<img width="1414" alt="image-20240205200400661" src="https://github.com/Harry-zklcdc/go-proxy-bingai/assets/21104213/743fbfc9-d60a-4cd9-abd6-1d4df8bf8201">

然后重启 Space

<img width="1415" alt="image-20240205200419465" src="https://github.com/Harry-zklcdc/go-proxy-bingai/assets/21104213/bcb0dddb-90d7-49fc-90b4-cac552c506ae">

> PS：这样设置完之后, 可以将空间设置为`Private`, 并且不影响访问

</details>

---

### CodeSandbox

> [!NOTE]
>
> 需要重新进行 [CodeSandbox 部署](https://github.com/Harry-zklcdc/go-proxy-bingai/wiki/CodeSandbox-%E9%83%A8%E7%BD%B2), 以更新代码支持

进入 CodeSandbox 的 DevBox

依次点击 左上角 「CodeSandbox」图标 ->「Restart Devbox」-> 「Settings」-> 「Env Variables」

![image](https://github.com/Harry-zklcdc/go-proxy-bingai/assets/21104213/fa6e4d38-397d-4d77-8e03-b4dd93683f01)

Name 输入 `CF_ZERO_TRUST_TOKEN`
Value 输入 `eyJhIjoiM2ZjNjQ4xxxxxxxxxxxxxxx`

> 注意 `eyJhIjoiM2ZjNjQ4xxxxxxxxxxxxxxx` 为刚才复制的 Token 值

![image](https://github.com/Harry-zklcdc/go-proxy-bingai/assets/21104213/ccecc199-184c-47a7-b71a-e6de453b0150)

保存重启 Devbox 即可

![image](https://github.com/Harry-zklcdc/go-proxy-bingai/assets/21104213/7e6770d4-f261-4212-aac9-40c55bb8ae39)

## 4. 配置域名

即可看到隧道上线

<img width="1415" alt="image-20240205201010176" src="https://github.com/Harry-zklcdc/go-proxy-bingai/assets/21104213/5bb9b2f7-9660-4fd0-bf78-52bdc0e2a686">

然后在下一个页面中填入如图所示内容

> `Subdomain` 是指子域名, 自定义
>
> `Domain`选择绑定到 Cloudflare 的域名, 自定义
>
> `Type`选择 `HTTP`, 不要改
>
> `URL` 填入 `localhost:8080`, 不要改 (**图是旧图, 一定要填 8080**)

<img width="1414" alt="image-20240205201111044" src="https://github.com/Harry-zklcdc/go-proxy-bingai/assets/21104213/ce637bd4-a69b-4c62-bc0f-ec4bd907dabd">

然后就可以用刚才填入的域名去访问了, 如图所示

<img width="1414" alt="image-20240205201412188" src="https://github.com/Harry-zklcdc/go-proxy-bingai/assets/21104213/429c9510-7ecc-492d-a9e6-3457bc16d36f">

<img width="1290" alt="image-20240205201503278" src="https://github.com/Harry-zklcdc/go-proxy-bingai/assets/21104213/df1cc874-d8c9-4282-8ce9-c3ccb2729108">

**Comment by Harry-zklcdc** [comment]
2024.02.07 HuggingFace 封杀本项目

**Comment by Sunmoone** [comment]
大佬，./cloudflared tunnel --no-autoupdate run --token $CF_ZERO_TRUST_TOKEN这个语句执行后提示cloudflare不存在

**Reply by Harry-zklcdc** [reply]
`cd /app`了吗？

**Reply by Sunmoone** [reply]
cd /app
export CF_ZERO_TRUST_TOKEN=自己的token，执行后没有任何输出，再执行/app/cloudflared tunnel --no-autoupdate run --token $CF_ZERO_TRUST_TOKEN，提示zsh: no such file or directory: /app/cloudflared，是要先安装cloud flare吗

**Reply by A8Cl233395** [reply]
+1，还有求apikey的更换教程

**Reply by Harry-zklcdc** [reply]
看 [Wiki](https://github.com/Harry-zklcdc/go-proxy-bingai/wiki/%E7%8E%AF%E5%A2%83%E5%8F%98%E9%87%8F#api-key)

**Reply by Harry-zklcdc** [reply]
这个我之后改一下教程，现在更简单了

**Reply by mountainguan** [reply]
/app/cloudflared tunnel --no-autoupdate run --token $CF_ZERO_TRUST_TOKEN
zsh: no such file or directory: /app/cloudflared

**Reply by Harry-zklcdc** [reply]
教程需要修改，我晚上改一下

**Reply by Harry-zklcdc** [reply]
已修改，重新配置即可

**Comment by ixxmu** [comment]
大佬如何让部署的 bing pass 人机验证服务也能添加在cf的zero trust？ 根据首页的提示部署在了codesanbox上，主要是发现自动过人机验证好像会一段时间断开，不知道换成自己的部署有没有区别

**Reply by Harry-zklcdc** [reply]
https://github.com/Harry-zklcdc/go-proxy-bingai/issues/352#issuecomment-1950985126
